### PR TITLE
[DiD] Rewritten S11 scenario

### PR DESCRIPTION
**Pull request by nemaara**
As the title states, a new scenario for S11 at @jostephd's suggestion.

### CODE REVIEW

**Comment by stevecotton** [comment]
Even with the runes, winning against the ghosts seems dependent on the RNG.

The heal-on-kill seems odd and unexplained during the pre-lich sequence.

**Comment by nemaara** [comment]
The ghost battle is more dependent on RNG than your average Wesnoth battle, but I balanced it up to a point where I thought it was winnable in every case as long as you play well enough. Also, none of the playtesters have complained about it yet on the latest version.

I'll let you think about why the heal-on-kill exists. :p

(If other people think it's a problem, I will rebalance it to be without the heal on kill, but I don't intend to explain it more obviously in game).

**Comment by jostephd** [comment]
> (If other people think it's a problem, I will rebalance it to be without the heal on kill, but I don't intend to explain it more obviously in game).

No argument about not explaining it in-game, but I think it's fair to ask for it to be explained on the PR and/or in source code comments.

Also, in the future could you please summarize the changes in your commit messages? Not just "updated gameplay" but also a high level (bullet points or paragraphs) summary of the changes, so other developers and playtesters would be able to follow the changes more easily. This is particularly important now, when the campaign is in testing by a wider audience.

**Comment by nemaara** [comment]
The in-scenario ritual is a metaphor for Malin's mind, or you could also interpret it as occurring inside his head as well. The three aspects of Malin Keshar that he fights represent aspects of "life", and he is actually the one taking on the role of "death". Just like undead (ghosts, necrophages, liches) siphon energy from the living to sustain themselves, so does Malin in the ritual (except that the roles are reversed). I suppose from that perspective, it would make sense for it to be a heal-on-hit instead of a heal on kill, but I found the on kill to provide for better gameplay so I chose that one.

Yes, I can write small changelogs for larger commits in the future.

File: data/campaigns/Descent_Into_Darkness/scenarios/11_Descent_into_Darkness.cfg [hunk 1]
```diff
@@ -391,419 +521,3071 @@
...
+                name=scenario_stage
+                numerical_equals=2
+            [/variable]
+        [/filter_condition]
+
+        [message]
+            speaker=narrator
+            # wmllint: local spelling Zephryn
+            message= _ "<s><i>Welcome to Zephryn’s laboratory. Please be mindful of ongoing experiments during your visit.</i></s>
+<big><i><b>This territory is the domain of the mighty Xanthric. All unwelcome intruders shall die.</b></i></big>"
```
Comment (Wedge009) [inline data/campaigns/Descent_Into_Darkness/scenarios/11_Descent_into_Darkness.cfg:1941]: By definition intruders are unwelcome... but I take it this tautology is deliberate as a sort of joke?

**Pull request merged**